### PR TITLE
Add class-based-sharding-spring-namespace-jpa-example for class-based-sharding-algorithm-example

### LINKED ISSUE #15715: Add `class-based-sharding-spring-namespace-jpa-example` for `class-based-sharding-algorithm-example`
Hi community,

### Background

[class-based-sharding-algorithm-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example) is an example of a custom algorithm that demonstrates how to integrate a user-defined sharding algorithm and declare the configuration using `CLASS_BASED` (avoid using SPI extensions)

### Aim

Add `class-based-sharding-spring-namespace-jpa-example` for `class-based-sharding-algorithm-example`

### Basic Qualifications

- Java
- Maven

### Example FYI

- [sharding-spring-namespace-jpa-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/sharding-example/sharding-spring-namespace-jpa-example)

### PR DESCRIPTION
**Pull request by Rohitpandey357**
Fixes #15715 .

### CODE REVIEW

**Comment by Rohitpandey357** [comment]
Hey @RaigorJiang, can you please take a look?

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases-tables.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    <bean id="demo_ds_1" class="com.zaxxer.hikari.HikariDataSource" destroy-method="close">
+        <property name="driverClassName" value="com.mysql.jdbc.Driver"/>
+        <property name="jdbcUrl" value="jdbc:mysql://localhost:3306/demo_ds_1?serverTimezone=UTC&amp;useSSL=false&amp;useUnicode=true&amp;characterEncoding=UTF-8"/>
+        <property name="username" value="root"/>
+        <property name="password" value=""/>
+        <property name="maximumPoolSize" value="16"/>
+    </bean>
+
+    <sharding:sharding-algorithm id="databaseAlgorithm" type="DATASOURCE_SPI_BASED" />
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases-tables.xml:73]: Hi @Rohitpandey357 
In this example, the type of `sharding-algorithm` should be "CLASS_BASED", NOT SPI.
And the props of `sharding-algorithm` should use the algorithm class you defined.
please refer to:
https://github.com/apache/shardingsphere/blob/master/examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-mybatis-example/src/main/resources/META-INF/application-sharding-databases-tables.xml

Comment (Rohitpandey357) [reply examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases-tables.xml]: Hi @RaigorJiang , I have modified the files as per your recommendations. Please have a look.

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases.xml [hunk 2]
```diff
@@ -92,5 +77,20 @@
...
     
-    <shardingsphere:data-source id="shardingDataSource" data-source-names="demo_ds_0, demo_ds_1" rule-refs="shardingRule" />
+    <shardingsphere:data-source id="shardingDataSource" database-name="sharding-databases" data-source-names="demo_ds_0, demo_ds_1" rule-refs="shardingRule" />
+    
+    <bean id="transactionManager" class="org.springframework.jdbc.datasource.DataSourceTransactionManager">
+        <property name="dataSource" ref="shardingDataSource" />
+    </bean>
+    <tx:annotation-driven />
+    
+    <bean id="sqlSessionFactory" class="org.mybatis.spring.SqlSessionFactoryBean">
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases.xml:87]: JPA example does not require sqlSessionFactory configuration for MyBatis, please check it.

Comment (Rohitpandey357) [reply examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases.xml:87]: Should I just remove this part? Or add something else instead?

Comment (Rohitpandey357) [reply examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-namespace-jpa-example/src/main/resources/META-INF/application-sharding-databases.xml]: Hey @RaigorJiang , I removed the mybatis config and pushed again.

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
@Rohitpandey357  Merged, thanks for your contribution.

**Pull request merged**